### PR TITLE
Improve load_area functionality and error report

### LINKED ISSUE #250: Misleading error when area file doesn't exist
#### Code Sample, a minimal, complete, and verifiable piece of code

```python
# Your code here
import pyresample.area_config
pyresample.area_config.load_area('file_that_does_not_exist.yaml', 'ease_sh')
```
#### Problem description
The above produces an obscure error when the file to load doesn't exist:
```python-traceback
Traceback (most recent call last):
  File "c:/Users/amsoe/OneDrive/Desktop/python/Untitled-1.py", line 16, in <module>
    area_def = load_area('areas.yaml', 'ease_sh')
  File "C:\Users\amsoe\AppData\Local\Programs\Python\Python38\lib\site-packages\pyresample\area_config.py", line 75, in load_area
    area_list = parse_area_file(area_file_name, *regions)
  File "C:\Users\amsoe\AppData\Local\Programs\Python\Python38\lib\site-packages\pyresample\area_config.py", line 104, in parse_area_file
    return _parse_yaml_area_file(area_file_name, *regions)
  File "C:\Users\amsoe\AppData\Local\Programs\Python\Python38\lib\site-packages\pyresample\area_config.py", line 139, in _parse_yaml_area_file
    area_dict = _read_yaml_area_file_content(area_file_name)
  File "C:\Users\amsoe\AppData\Local\Programs\Python\Python38\lib\site-packages\pyresample\area_config.py", line 123, in _read_yaml_area_file_content
    area_dict = recursive_dict_update(area_dict, tmp_dict)
  File "C:\Users\amsoe\AppData\Local\Programs\Python\Python38\lib\site-packages\pyresample\utils\_init_.py", line 234, in recursive_dict_update
    for k, v in u.items():
AttributeError: 'str' object has no attribute 'items' 
```

#### Expected Output
A nice error that tells me the file I want to load doesn't exist.

### PR DESCRIPTION
**Pull request by gerritholl**
In area_config.load_area, add the functionality to load from a
pathlib.Path object or a stream.  Deprecate the functionality to load
from a string directly --- temporarily, it will assume a string
descirbes a yaml file if it contains a newline, and a path otherwise.
If a file does not exist or is not readable, leave it to the open
function to raise the appropriate exception.

Closes #250.

 - [x] Closes #250 
 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

File: pyresample/test/test_utils.py [hunk 1]
```diff
@@ -101,6 +103,13 @@ def test_load_area(self):
 Area extent: (-5326849.0625, -5326849.0625, 5326849.0625, 5326849.0625)""".format(projection)
         self.assertEqual(nh_str, ease_nh.__str__())
 
+    def test_area_file_not_found_exception(self):
+        from pyresample.area_config import load_area
+        self.assertRaises(FileNotFoundError, load_area,
```
[the remark this answers is not in the document]
Comment (gerritholl) [reply pyresample/test/test_utils.py]: It doesn't say that on my system.

[the remark this answers is not in the document]
Comment (mraspaud) [reply pyresample/test/test_utils.py]: That's a python 2/3 thing. IIRC FileNotFoundError is python3 only. I wonder if stickler is running python 2 tests only

**Comment by gerritholl** [comment]
There are tests failing in a part of the code I didn't touch at all.  I don't know why they are failing.  They run fine on my machine.  Why could that be?

**Comment by djhoese** [comment]
Is "stream" the proper name for this? I thought it was more common to use "file-like object". After re-reading the `io` documentation I see they use `stream` a lot. :thinking: 

I'm sad to see string support go.

**Comment by gerritholl** [comment]
Is there an alternative solution that prevents the problem in #250 without making string support go?  In theory one could make my ad-hoc "any string with a newline cannot be a sane filename" permanent.  I don't know enough about the definition of yaml to tell if a (machine generated?) yaml file could exist without newlines.

I don't know if there is a difference between streams and file-like objects.

**Comment by mraspaud** [comment]
How about having an explicit `load_areas_from_string` method that is named in the deprecation warning ?
It can be as simple as 
```python
def load_areas_from_string(self, the_string, *args, **kwargs):
    return self.load_areas(StringIO(the_string), *args, **kwargs)
```

**Comment by djhoese** [comment]
I was thinking go the other way, "what about about this string should mean it is a filename"? Does it have a `yml` or `yaml` extension? Or if it starts with a `/` or `.` then it is a path. Otherwise maybe an explicit check `if not os.path.isfile(X):` followed by a `try/except` of parsing the yaml...eh that's basically what we have now.

**Comment by gerritholl** [comment]
I find testing if a string looks like a filename rather error-prone.  "tmpOEuao234b" is a filename, and in a unit testing framework not even one an unexpected one, but what positive test could confirm this?  I prefer @mraspaud suggestion and have different functions, because explicit is better than implicit.

**Comment by djhoese** [comment]
Good enough. I'd rather not having the separate method and just require people to make a `StringIO` object if they really need to pass a string.

**Comment by djhoese** [comment]
I fixed the libtiff issue (or rather worked around it) in #257. Merging this since the relevant tests pass.

**Pull request merged**